single_kernel_mongo/managers/sharding.py

```diff
@@ -545,29 +544,11 @@ def assert_pass_hook_checks(self, relation: Relation, is_leaving: bool = False)
                 "Config-server never set up, no need to process broken event."
             )
 
-        shard_has_tls, config_server_has_tls = self.tls_status()
-        match (shard_has_tls, config_server_has_tls):
-            case False, True:
-                self.state.statuses.add(
-                    ShardStatuses.REQUIRES_TLS.value, scope="unit", component=self.name
-                )
-                raise DeferrableFailedHookChecksError(
-                    "Config-Server uses TLS but shard does not. Please synchronise encryption method."
-                )
-            case True, False:
-                self.state.statuses.add(
-                    ShardStatuses.REQUIRES_NO_TLS.value, scope="unit", component=self.name
-                )
-                raise DeferrableFailedHookChecksError(
-                    "Shard uses TLS but config-server does not. Please synchronise encryption method."
-                )
-            case _:
-                pass
-
-        if not self.is_ca_compatible():
-            raise DeferrableFailedHookChecksError(
-                "Shard is integrated to a different CA than the config server. Please use the same CA for all cluster components.",
-            )
+        tls_status = self.get_tls_status()
+        if tls_status:
+            self.state.statuses.add(tls_status, scope="unit", component=self.name)
+            exception_msg = f"{tls_status.message} {tls_status.action}"
+            raise DeferrableFailedHookChecksError(exception_msg)
```

Comment:
Nit: walrus operator
```suggestion
        if tls_status := self.get_tls_status():
            self.state.statuses.add(tls_status, scope="unit", component=self.name)
            exception_msg = f"{tls_status.message} {tls_status.action}"
            raise DeferrableFailedHookChecksError(exception_msg)
```